sdk/ml/azure-ai-ml/azure/ai/ml/operations/_code_operations.py

```diff
@@ -92,9 +100,36 @@ def create_or_update(self, code: Code) -> Code:
                     registry=self._registry_name,
                     body=get_asset_body_for_registry_storage(self._registry_name, "codes", name, version),
                 )
-            code, _ = _check_and_upload_path(
-                artifact=code, asset_operations=self, sas_uri=sas_uri, artifact_type=ErrorTarget.CODE
-            )
+                code, _ = _check_and_upload_path(
+                    artifact=code,
+                    asset_operations=self,
+                    sas_uri=sas_uri,
+                    artifact_type=ErrorTarget.CODE
+                )
+            else:
+                path, ignore_file, asset_hash = _get_snapshot_path_info(code)
+                workspace_info = self._datastore_operation._service_client.workspaces.get(
```

Comment:
Consider passing _operation_container into the CodeOperation class. Depending on a private property of the DatastoreOperation is risky.